src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        joinedStr += `${this[i]}`;
```

Comment:
```suggestion
        joinedStr += this[i];
```

Reply:
What happens if we will add `string` to `number`, what type result will have?
For example 'Some text'+4=?

Reply:
Oh, yep it works, I was a little bit confused by another programming language which I learn previously. But actually, is it a good practice to add a number to a string?